### PR TITLE
Handles hierarchy

### PR DESCRIPTION
**Pull request by mario-fuentes**

### CODE REVIEW

**Comment by radekdoulik** [comment]
Could you please squash the commits together?

File: tools/Generator/Generator.cs [hunk 1]
```diff
@@ -94,7 +95,7 @@ string GetTypeCsName (string name, string typeName = "type")
 		{
 			if (typesTranslation.ContainsKey (name))
 				return typesTranslation [name];
-
+
```
Comment (radekdoulik) [inline tools/Generator/Generator.cs]: Please don't change white space.

File: tools/Generator/Generator.cs [hunk 2]
```diff
@@ -187,17 +192,9 @@ bool WriteEnum (XElement enumElement)
 
 			var enumsElement = values.First ();
 			if (enumsElement.Attribute ("type") != null && enumsElement.Attribute ("type").Value == "bitmask") {
-				string suffix = null;
-				foreach (var ext in extensions)
-					if (csName.EndsWith (ext.Key)) {
-						suffix = ext.Key + suffix;
-						csName = csName.Substring (0, csName.Length - ext.Key.Length);
-					}
+
```
Comment (radekdoulik) [inline tools/Generator/Generator.cs]: This will regress the Enums.cs. See CompositeAlphaFlagsKHR for example.

File: tools/Generator/Generator.cs [hunk 3]
```diff
@@ -187,17 +192,9 @@ bool WriteEnum (XElement enumElement)
...
-						suffix = ext.Key + suffix;
-						csName = csName.Substring (0, csName.Length - ext.Key.Length);
-					}
+				
 				writer.WriteLine ("\t[Flags]");
-				if (csName.EndsWith ("FlagBits"))
-					csName = csName.Substring (0, csName.Length - 4) + "s";
-				if (suffix != null)
-					csName += suffix;
+				csName = csName.Replace ("FlagBits", "Flags");
```
Comment (radekdoulik) [inline tools/Generator/Generator.cs]: Same as above.

File: tools/Generator/Generator.cs (hunk 2 above)
Comment (mario-fuentes) [inline tools/Generator/Generator.cs]: What enum name do you want, CompositeAlphaFlagsKHR  or CompositeAlphaFlagsKhr??

Comment (radekdoulik) [inline tools/Generator/Generator.cs]: CompositeAlphaFlagsKhr, we cannot break the prefix/suffix logic though

Comment (mario-fuentes) [inline tools/Generator/Generator.cs]: So, I just moved the code who replace the extension suffix to the GetCsTypeName method, current code generate "CompositeAlphaFlagsKHR", my code generate "CompositeAlphaFlagsKhr".  The enum's members aren't generated right, I fixed it locally (removed the enum name prefix), I will commit it soon.

File: tools/Generator/Generator.cs [hunk 4]
```diff
@@ -396,8 +393,19 @@ bool WriteHandle(XElement handleElement)
...
+
+			if (handleElement.Attribute ("parent") != null)
+				parent = handleElement.Attribute ("parent").Value
+					.Split (',')
+					.Select (pName => GetTypeCsName (pName))
+					.Aggregate ((a, p) => string.Format ("{0}, {1}", a, p));
+			
+			writer.WriteLine ("\tpublic class {0}{1}\n\t{{\n\t}}",
+				csName,
+				parent == null ? string.Empty : string.Format (" : {0}", parent));
```
Comment (mario-fuentes) [inline tools/Generator/Generator.cs]: @radekdoulik This is not good, the Vulkan API suppose multi-hierarchy (VkDisplayModeKHR).

**Comment by mario-fuentes** [comment]
I'm closing this PR, I will create a new one soon

File: tools/Generator/Generator.cs (hunk 4 above)
Comment (radekdoulik) [inline tools/Generator/Generator.cs]: VkDisplayModeKHR is non-dispatchable handle, so it is 64bit integer, so I think we don't need to do anything about hierarchy.

right now we have all handles as pointers, which is wrong and we should take care about it at some point

**Pull request closed without merging**